web/react/components/mention_list.jsx

```diff
@@ -23,6 +23,11 @@ module.exports = React.createClass({
                 }
             }
         );
+        $(document).click(function() {
+            if($('#'+self.props.id).length && $('#'+self.props.id).get(0) !== $(':focus').get(0)) {
```

Comment:
Here, self.props.id represents whatever text box this particular mentions list is linked to. We reference things in JQuery by doing $(<insert identifier here>). 
First we check the length to ensure that the element exists (due to the virtual DOM of react, this can be a problem sometimes, don't want to be throwing around errors). Then, we get the internal DOM object with get(0), so we can go from the JQuery identification to the actual DOM. This is necessary as a facet of the language and how it selects parts of the page.
We also get the DOM node of whatever is in focus, as represented by $(':focus').get(0). This returns whatever is the current element that the user has selected (eg if the user is focused on a text box, it will have a blinking cursor). So here we specify "when that click event is triggered, check if the current active element is the text box I'm associated with." (I'm referring to the object MetionList, not the committer).